Libs/Visualization/VTK/Core/vtkDiscretizableColorTransferChart.cpp

```diff
@@ -403,44 +425,84 @@ void vtkDiscretizableColorTransferChart::SetCurrentControlPointColor(
 // ----------------------------------------------------------------------------
 void vtkDiscretizableColorTransferChart::SetDataRange(double min, double max)
 {
+  if (min == this->DataRange[0]
+   && max == this->DataRange[1])
+  {
+    return;
+  }
+
   this->DataRange[0] = min;
   this->DataRange[1] = max;
-}
 
-// ----------------------------------------------------------------------------
-double* vtkDiscretizableColorTransferChart::GetDataRange()
-{
-  return this->DataRange;
+#ifdef DEBUG_RANGE
+  std::cout << "DEBUG_RANGE chart data range = " << this->DataRange[0]
+            << " " << this->DataRange[1] << std::endl;
+#endif
```

Comment:
I'm surprised it does not trigger any refresh/computation

Reply:
The data range is used to rescale or center the current range, which is triggered by the user. I think we don't want to rescale everytime the data range changes (new input for histogram). Tell me if you don't agree.

Reply:
i guess that works too